tools/ci.sh

```diff
@@ -317,19 +317,54 @@ function ci_qemu_setup_rv32 {
     qemu-system-riscv32 --version
 }
 
-function ci_qemu_build_arm {
+function ci_qemu_build_arm_prepare {
     make ${MAKEOPTS} -C mpy-cross
     make ${MAKEOPTS} -C ports/qemu submodules
+}
+
+function ci_qemu_build_arm_bigendian {
     make ${MAKEOPTS} -C ports/qemu CFLAGS_EXTRA=-DMP_ENDIANNESS_BIG=1
-    make ${MAKEOPTS} -C ports/qemu clean
-    make ${MAKEOPTS} -C ports/qemu test_full
+}
+
+function ci_qemu_build_arm_sabrelite {
     make ${MAKEOPTS} -C ports/qemu BOARD=SABRELITE test_full
+}
+
+function ci_qemu_build_arm_thumb {
+    make ${MAKEOPTS} -C ports/qemu test_full
 
     # Test building and running native .mpy with armv7m architecture.
     ci_native_mpy_modules_build armv7m
     make ${MAKEOPTS} -C ports/qemu test_natmod
 }
 
+function ci_qemu_build_arm {
```

Comment:
IMO, it's simpler (and definitely less code) to just have three functions `ci_qemu_build_arm_bigendian`, `ci_qemu_build_arm_sabrelite` and `ci_qemu_build_arm_thumb` and make those functions call `ci_qemu_build_arm_prepare` at the start of them.

Then the workflow can just call `ci_qemu_build_arm_${{ matrix.ci_func }}` directly (that's how stm32 workflow does it).

There's no need to support the old way of calling `ci_qemu_build_arm`.

Reply:
> There's no need to support the old way of calling `ci_qemu_build_arm`.

I know at least one person whose tooling would break due to this :)

But that's fine - I'll make the necessary changes.

Reply:
Haha, I see!

MicroPython has a very big user surface area.  I think it's just adding to our work if we add `tools/ci.sh` to that list of things we can't change.